// tpvm/generated/TPGrammarLexer.js
// Generated from /media/ecastro/Store/tpvm/codegen/TPGrammar.g4 by ANTLR 4.7
// jshint ignore: start
var antlr4 = require('antlr4/index');


var serializedATN = ["\u0003\u608b\ua72a\u8133\ub9ed\u417c\u3be7\u7786\u5964",
    "\u0002\u001f\u011d\b\u0001\u0004\u0002\t\u0002\u0004\u0003\t\u0003\u0004",
    "\u0004\t\u0004\u0004\u0005\t\u0005\u0004\u0006\t\u0006\u0004\u0007\t",
    "\u0007\u0004\b\t\b\u0004\t\t\t\u0004\n\t\n\u0004\u000b\t\u000b\u0004",
    "\f\t\f\u0004\r\t\r\u0004\u000e\t\u000e\u0004\u000f\t\u000f\u0004\u0010",
    "\t\u0010\u0004\u0011\t\u0011\u0004\u0012\t\u0012\u0004\u0013\t\u0013",
    "\u0004\u0014\t\u0014\u0004\u0015\t\u0015\u0004\u0016\t\u0016\u0004\u0017",
    "\t\u0017\u0004\u0018\t\u0018\u0004\u0019\t\u0019\u0004\u001a\t\u001a",
    "\u0004\u001b\t\u001b\u0004\u001c\t\u001c\u0004\u001d\t\u001d\u0004\u001e",
    "\t\u001e\u0004\u001f\t\u001f\u0004 \t \u0004!\t!\u0004\"\t\"\u0004#",
    "\t#\u0004$\t$\u0004%\t%\u0003\u0002\u0003\u0002\u0003\u0003\u0003\u0003",
    "\u0003\u0004\u0006\u0004Q\n\u0004\r\u0004\u000e\u0004R\u0003\u0004\u0003",
    "\u0004\u0007\u0004W\n\u0004\f\u0004\u000e\u0004Z\u000b\u0004\u0003\u0005",
    "\u0003\u0005\u0003\u0005\u0003\u0006\u0003\u0006\u0003\u0006\u0003\u0006",
    "\u0003\u0006\u0003\u0007\u0003\u0007\u0003\b\u0003\b\u0003\b\u0003\t",
    "\u0003\t\u0003\t\u0003\t\u0003\n\u0003\n\u0003\n\u0003\n\u0003\u000b",
    "\u0003\u000b\u0003\u000b\u0003\f\u0003\f\u0003\r\u0003\r\u0003\r\u0005",
    "\ry\n\r\u0003\u000e\u0003\u000e\u0003\u000f\u0003\u000f\u0003\u000f",
    "\u0005\u000f\u0080\n\u000f\u0003\u0010\u0003\u0010\u0003\u0010\u0005",
    "\u0010\u0085\n\u0010\u0003\u0011\u0003\u0011\u0003\u0011\u0003\u0012",
    "\u0003\u0012\u0003\u0012\u0003\u0013\u0003\u0013\u0003\u0014\u0003\u0014",
    "\u0003\u0015\u0003\u0015\u0003\u0016\u0003\u0016\u0003\u0017\u0003\u0017",
    "\u0003\u0017\u0003\u0017\u0003\u0017\u0003\u0017\u0003\u0017\u0003\u0017",
    "\u0003\u0017\u0005\u0017\u009e\n\u0017\u0003\u0018\u0006\u0018\u00a1",
    "\n\u0018\r\u0018\u000e\u0018\u00a2\u0003\u0018\u0003\u0018\u0003\u0018",
    "\u0003\u0018\u0006\u0018\u00a9\n\u0018\r\u0018\u000e\u0018\u00aa\u0003",
    "\u0018\u0003\u0018\u0003\u0018\u0003\u0018\u0005\u0018\u00b1\n\u0018",
    "\u0003\u0019\u0003\u0019\u0005\u0019\u00b5\n\u0019\u0003\u001a\u0003",
    "\u001a\u0003\u001b\u0003\u001b\u0003\u001b\u0007\u001b\u00bc\n\u001b",
    "\f\u001b\u000e\u001b\u00bf\u000b\u001b\u0003\u001c\u0003\u001c\u0003",
    "\u001d\u0003\u001d\u0003\u001e\u0003\u001e\u0003\u001f\u0003\u001f\u0003",
    " \u0007 \u00ca\n \f \u000e \u00cd\u000b \u0003 \u0003 \u0006 \u00d1",
    "\n \r \u000e \u00d2\u0003 \u0006 \u00d6\n \r \u000e \u00d7\u0003 \u0003",
    " \u0007 \u00dc\n \f \u000e \u00df\u000b \u0003 \u0006 \u00e2\n \r \u000e",
    " \u00e3\u0005 \u00e6\n \u0003!\u0003!\u0005!\u00ea\n!\u0003!\u0006!",
    "\u00ed\n!\r!\u000e!\u00ee\u0003\"\u0003\"\u0003\"\u0003\"\u0003#\u0003",
    "#\u0003#\u0003#\u0007#\u00f9\n#\f#\u000e#\u00fc\u000b#\u0003#\u0003",
    "#\u0003#\u0003#\u0003#\u0003$\u0003$\u0003$\u0003$\u0007$\u0107\n$\f",
    "$\u000e$\u010a\u000b$\u0003$\u0005$\u010d\n$\u0003$\u0003$\u0003$\u0003",
    "$\u0003%\u0003%\u0003%\u0003%\u0007%\u0117\n%\f%\u000e%\u011a\u000b",
    "%\u0003%\u0003%\u0003\u00fa\u0002&\u0003\u0003\u0005\u0004\u0007\u0005",
    "\t\u0006\u000b\u0007\r\b\u000f\t\u0011\n\u0013\u000b\u0015\f\u0017\r",
    "\u0019\u000e\u001b\u000f\u001d\u0010\u001f\u0011!\u0012#\u0013%\u0014",
    "\'\u0015)\u0016+\u0017-\u0018/\u00191\u001a3\u001b5\u00027\u00029\u0002",
    ";\u0002=\u0002?\u0002A\u0002C\u001cE\u001dG\u001eI\u001f\u0003\u0002",
    "\r\u0004\u0002,,\u00d9\u00d9\u0004\u000211\u00f9\u00f9\u0004\u0002/",
    "/\u2214\u2214\u0005\u0002C\\aac|\u0003\u00022;\u0005\u00022;CHch\u0003",
    "\u000223\u0004\u0002GGgg\u0004\u0002--//\u0005\u0002\u000b\f\u000e\u000f",
    "\"\"\u0004\u0002\f\f\u000f\u000f\u0002\u0130\u0002\u0003\u0003\u0002",
    "\u0002\u0002\u0002\u0005\u0003\u0002\u0002\u0002\u0002\u0007\u0003\u0002",
    "\u0002\u0002\u0002\t\u0003\u0002\u0002\u0002\u0002\u000b\u0003\u0002",
    "\u0002\u0002\u0002\r\u0003\u0002\u0002\u0002\u0002\u000f\u0003\u0002",
    "\u0002\u0002\u0002\u0011\u0003\u0002\u0002\u0002\u0002\u0013\u0003\u0002",
    "\u0002\u0002\u0002\u0015\u0003\u0002\u0002\u0002\u0002\u0017\u0003\u0002",
    "\u0002\u0002\u0002\u0019\u0003\u0002\u0002\u0002\u0002\u001b\u0003\u0002",
    "\u0002\u0002\u0002\u001d\u0003\u0002\u0002\u0002\u0002\u001f\u0003\u0002",
    "\u0002\u0002\u0002!\u0003\u0002\u0002\u0002\u0002#\u0003\u0002\u0002",
    "\u0002\u0002%\u0003\u0002\u0002\u0002\u0002\'\u0003\u0002\u0002\u0002",
    "\u0002)\u0003\u0002\u0002\u0002\u0002+\u0003\u0002\u0002\u0002\u0002",
    "-\u0003\u0002\u0002\u0002\u0002/\u0003\u0002\u0002\u0002\u00021\u0003",
    "\u0002\u0002\u0002\u00023\u0003\u0002\u0002\u0002\u0002C\u0003\u0002",
    "\u0002\u0002\u0002E\u0003\u0002\u0002\u0002\u0002G\u0003\u0002\u0002",
    "\u0002\u0002I\u0003\u0002\u0002\u0002\u0003K\u0003\u0002\u0002\u0002",
    "\u0005M\u0003\u0002\u0002\u0002\u0007P\u0003\u0002\u0002\u0002\t[\u0003",
    "\u0002\u0002\u0002\u000b^\u0003\u0002\u0002\u0002\rc\u0003\u0002\u0002",
    "\u0002\u000fe\u0003\u0002\u0002\u0002\u0011h\u0003\u0002\u0002\u0002",
    "\u0013l\u0003\u0002\u0002\u0002\u0015p\u0003\u0002\u0002\u0002\u0017",
    "s\u0003\u0002\u0002\u0002\u0019x\u0003\u0002\u0002\u0002\u001bz\u0003",
    "\u0002\u0002\u0002\u001d\u007f\u0003\u0002\u0002\u0002\u001f\u0084\u0003",
    "\u0002\u0002\u0002!\u0086\u0003\u0002\u0002\u0002#\u0089\u0003\u0002",
    "\u0002\u0002%\u008c\u0003\u0002\u0002\u0002\'\u008e\u0003\u0002\u0002",
    "\u0002)\u0090\u0003\u0002\u0002\u0002+\u0092\u0003\u0002\u0002\u0002",
    "-\u009d\u0003\u0002\u0002\u0002/\u00b0\u0003\u0002\u0002\u00021\u00b2",
    "\u0003\u0002\u0002\u00023\u00b6\u0003\u0002\u0002\u00025\u00b8\u0003",
    "\u0002\u0002\u00027\u00c0\u0003\u0002\u0002\u00029\u00c2\u0003\u0002",
    "\u0002\u0002;\u00c4\u0003\u0002\u0002\u0002=\u00c6\u0003\u0002\u0002",
    "\u0002?\u00e5\u0003\u0002\u0002\u0002A\u00e7\u0003\u0002\u0002\u0002",
    "C\u00f0\u0003\u0002\u0002\u0002E\u00f4\u0003\u0002\u0002\u0002G\u0102",
    "\u0003\u0002\u0002\u0002I\u0112\u0003\u0002\u0002\u0002KL\u0007*\u0002",
    "\u0002L\u0004\u0003\u0002\u0002\u0002MN\u0007+\u0002\u0002N\u0006\u0003",
    "\u0002\u0002\u0002OQ\u00059\u001d\u0002PO\u0003\u0002\u0002\u0002QR",
    "\u0003\u0002\u0002\u0002RP\u0003\u0002\u0002\u0002RS\u0003\u0002\u0002",
    "\u0002SX\u0003\u0002\u0002\u0002TW\u00057\u001c\u0002UW\u00059\u001d",
    "\u0002VT\u0003\u0002\u0002\u0002VU\u0003\u0002\u0002\u0002WZ\u0003\u0002",
    "\u0002\u0002XV\u0003\u0002\u0002\u0002XY\u0003\u0002\u0002\u0002Y\b",
    "\u0003\u0002\u0002\u0002ZX\u0003\u0002\u0002\u0002[\\\u0007k\u0002\u0002",
    "\\]\u0007h\u0002\u0002]\n\u0003\u0002\u0002\u0002^_\u0007g\u0002\u0002",
    "_`\u0007n\u0002\u0002`a\u0007u\u0002\u0002ab\u0007g\u0002\u0002b\f\u0003",
    "\u0002\u0002\u0002cd\u0007?\u0002\u0002d\u000e\u0003\u0002\u0002\u0002",
    "ef\u0007q\u0002\u0002fg\u0007t\u0002\u0002g\u0010\u0003\u0002\u0002",
    "\u0002hi\u0007c\u0002\u0002ij\u0007p\u0002\u0002jk\u0007f\u0002\u0002",
    "k\u0012\u0003\u0002\u0002\u0002lm\u0007p\u0002\u0002mn\u0007q\u0002",
    "\u0002no\u0007v\u0002\u0002o\u0014\u0003\u0002\u0002\u0002pq\u0007?",
    "\u0002\u0002qr\u0007?\u0002\u0002r\u0016\u0003\u0002\u0002\u0002st\u0007",
    ">\u0002\u0002t\u0018\u0003\u0002\u0002\u0002uv\u0007>\u0002\u0002vy",
    "\u0007?\u0002\u0002wy\u0007\u2266\u0002\u0002xu\u0003\u0002\u0002\u0002",
    "xw\u0003\u0002\u0002\u0002y\u001a\u0003\u0002\u0002\u0002z{\u0007@\u0002",
    "\u0002{\u001c\u0003\u0002\u0002\u0002|}\u0007@\u0002\u0002}\u0080\u0007",
    "?\u0002\u0002~\u0080\u0007\u2267\u0002\u0002\u007f|\u0003\u0002\u0002",
    "\u0002\u007f~\u0003\u0002\u0002\u0002\u0080\u001e\u0003\u0002\u0002",
    "\u0002\u0081\u0082\u0007#\u0002\u0002\u0082\u0085\u0007?\u0002\u0002",
    "\u0083\u0085\u0007\u2262\u0002\u0002\u0084\u0081\u0003\u0002\u0002\u0002",
    "\u0084\u0083\u0003\u0002\u0002\u0002\u0085 \u0003\u0002\u0002\u0002",
    "\u0086\u0087\u0007-\u0002\u0002\u0087\u0088\u0007-\u0002\u0002\u0088",
    "\"\u0003\u0002\u0002\u0002\u0089\u008a\u0007/\u0002\u0002\u008a\u008b",
    "\u0007/\u0002\u0002\u008b$\u0003\u0002\u0002\u0002\u008c\u008d\t\u0002",
    "\u0002\u0002\u008d&\u0003\u0002\u0002\u0002\u008e\u008f\t\u0003\u0002",
    "\u0002\u008f(\u0003\u0002\u0002\u0002\u0090\u0091\u0007-\u0002\u0002",
    "\u0091*\u0003\u0002\u0002\u0002\u0092\u0093\t\u0004\u0002\u0002\u0093",
    ",\u0003\u0002\u0002\u0002\u0094\u0095\u0007v\u0002\u0002\u0095\u0096",
    "\u0007t\u0002\u0002\u0096\u0097\u0007w\u0002\u0002\u0097\u009e\u0007",
    "g\u0002\u0002\u0098\u0099\u0007h\u0002\u0002\u0099\u009a\u0007c\u0002",
    "\u0002\u009a\u009b\u0007n\u0002\u0002\u009b\u009c\u0007u\u0002\u0002",
    "\u009c\u009e\u0007g\u0002\u0002\u009d\u0094\u0003\u0002\u0002\u0002",
    "\u009d\u0098\u0003\u0002\u0002\u0002\u009e.\u0003\u0002\u0002\u0002",
    "\u009f\u00a1\u00059\u001d\u0002\u00a0\u009f\u0003\u0002\u0002\u0002",
    "\u00a1\u00a2\u0003\u0002\u0002\u0002\u00a2\u00a0\u0003\u0002\u0002\u0002",
    "\u00a2\u00a3\u0003\u0002\u0002\u0002\u00a3\u00b1\u0003\u0002\u0002\u0002",
    "\u00a4\u00a5\u00072\u0002\u0002\u00a5\u00a6\u0007z\u0002\u0002\u00a6",
    "\u00a8\u0003\u0002\u0002\u0002\u00a7\u00a9\u0005;\u001e\u0002\u00a8",
    "\u00a7\u0003\u0002\u0002\u0002\u00a9\u00aa\u0003\u0002\u0002\u0002\u00aa",
    "\u00a8\u0003\u0002\u0002\u0002\u00aa\u00ab\u0003\u0002\u0002\u0002\u00ab",
    "\u00b1\u0003\u0002\u0002\u0002\u00ac\u00ad\u00072\u0002\u0002\u00ad",
    "\u00ae\u0007d\u0002\u0002\u00ae\u00af\u0003\u0002\u0002\u0002\u00af",
    "\u00b1\u0005=\u001f\u0002\u00b0\u00a0\u0003\u0002\u0002\u0002\u00b0",
    "\u00a4\u0003\u0002\u0002\u0002\u00b0\u00ac\u0003\u0002\u0002\u0002\u00b1",
    "0\u0003\u0002\u0002\u0002\u00b2\u00b4\u0005? \u0002\u00b3\u00b5\u0005",
    "A!\u0002\u00b4\u00b3\u0003\u0002\u0002\u0002\u00b4\u00b5\u0003\u0002",
    "\u0002\u0002\u00b52\u0003\u0002\u0002\u0002\u00b6\u00b7\u00055\u001b",
    "\u0002\u00b74\u0003\u0002\u0002\u0002\u00b8\u00bd\u00057\u001c\u0002",
    "\u00b9\u00bc\u00057\u001c\u0002\u00ba\u00bc\u00059\u001d\u0002\u00bb",
    "\u00b9\u0003\u0002\u0002\u0002\u00bb\u00ba\u0003\u0002\u0002\u0002\u00bc",
    "\u00bf\u0003\u0002\u0002\u0002\u00bd\u00bb\u0003\u0002\u0002\u0002\u00bd",
    "\u00be\u0003\u0002\u0002\u0002\u00be6\u0003\u0002\u0002\u0002\u00bf",
    "\u00bd\u0003\u0002\u0002\u0002\u00c0\u00c1\t\u0005\u0002\u0002\u00c1",
    "8\u0003\u0002\u0002\u0002\u00c2\u00c3\t\u0006\u0002\u0002\u00c3:\u0003",
    "\u0002\u0002\u0002\u00c4\u00c5\t\u0007\u0002\u0002\u00c5<\u0003\u0002",
    "\u0002\u0002\u00c6\u00c7\t\b\u0002\u0002\u00c7>\u0003\u0002\u0002\u0002",
    "\u00c8\u00ca\u00059\u001d\u0002\u00c9\u00c8\u0003\u0002\u0002\u0002",
    "\u00ca\u00cd\u0003\u0002\u0002\u0002\u00cb\u00c9\u0003\u0002\u0002\u0002",
    "\u00cb\u00cc\u0003\u0002\u0002\u0002\u00cc\u00ce\u0003\u0002\u0002\u0002",
    "\u00cd\u00cb\u0003\u0002\u0002\u0002\u00ce\u00d0\u00070\u0002\u0002",
    "\u00cf\u00d1\u00059\u001d\u0002\u00d0\u00cf\u0003\u0002\u0002\u0002",
    "\u00d1\u00d2\u0003\u0002\u0002\u0002\u00d2\u00d0\u0003\u0002\u0002\u0002",
    "\u00d2\u00d3\u0003\u0002\u0002\u0002\u00d3\u00e6\u0003\u0002\u0002\u0002",
    "\u00d4\u00d6\u00059\u001d\u0002\u00d5\u00d4\u0003\u0002\u0002\u0002",
    "\u00d6\u00d7\u0003\u0002\u0002\u0002\u00d7\u00d5\u0003\u0002\u0002\u0002",
    "\u00d7\u00d8\u0003\u0002\u0002\u0002\u00d8\u00d9\u0003\u0002\u0002\u0002",
    "\u00d9\u00dd\u00070\u0002\u0002\u00da\u00dc\u00059\u001d\u0002\u00db",
    "\u00da\u0003\u0002\u0002\u0002\u00dc\u00df\u0003\u0002\u0002\u0002\u00dd",
    "\u00db\u0003\u0002\u0002\u0002\u00dd\u00de\u0003\u0002\u0002\u0002\u00de",
    "\u00e6\u0003\u0002\u0002\u0002\u00df\u00dd\u0003\u0002\u0002\u0002\u00e0",
    "\u00e2\u00059\u001d\u0002\u00e1\u00e0\u0003\u0002\u0002\u0002\u00e2",
    "\u00e3\u0003\u0002\u0002\u0002\u00e3\u00e1\u0003\u0002\u0002\u0002\u00e3",
    "\u00e4\u0003\u0002\u0002\u0002\u00e4\u00e6\u0003\u0002\u0002\u0002\u00e5",
    "\u00cb\u0003\u0002\u0002\u0002\u00e5\u00d5\u0003\u0002\u0002\u0002\u00e5",
    "\u00e1\u0003\u0002\u0002\u0002\u00e6@\u0003\u0002\u0002\u0002\u00e7",
    "\u00e9\t\t\u0002\u0002\u00e8\u00ea\t\n\u0002\u0002\u00e9\u00e8\u0003",
    "\u0002\u0002\u0002\u00e9\u00ea\u0003\u0002\u0002\u0002\u00ea\u00ec\u0003",
    "\u0002\u0002\u0002\u00eb\u00ed\u00059\u001d\u0002\u00ec\u00eb\u0003",
    "\u0002\u0002\u0002\u00ed\u00ee\u0003\u0002\u0002\u0002\u00ee\u00ec\u0003",
    "\u0002\u0002\u0002\u00ee\u00ef\u0003\u0002\u0002\u0002\u00efB\u0003",
    "\u0002\u0002\u0002\u00f0\u00f1\t\u000b\u0002\u0002\u00f1\u00f2\u0003",
    "\u0002\u0002\u0002\u00f2\u00f3\b\"\u0002\u0002\u00f3D\u0003\u0002\u0002",
    "\u0002\u00f4\u00f5\u00071\u0002\u0002\u00f5\u00f6\u0007,\u0002\u0002",
    "\u00f6\u00fa\u0003\u0002\u0002\u0002\u00f7\u00f9\u000b\u0002\u0002\u0002",
    "\u00f8\u00f7\u0003\u0002\u0002\u0002\u00f9\u00fc\u0003\u0002\u0002\u0002",
    "\u00fa\u00fb\u0003\u0002\u0002\u0002\u00fa\u00f8\u0003\u0002\u0002\u0002",
    "\u00fb\u00fd\u0003\u0002\u0002\u0002\u00fc\u00fa\u0003\u0002\u0002\u0002",
    "\u00fd\u00fe\u0007,\u0002\u0002\u00fe\u00ff\u00071\u0002\u0002\u00ff",
    "\u0100\u0003\u0002\u0002\u0002\u0100\u0101\b#\u0002\u0002\u0101F\u0003",
    "\u0002\u0002\u0002\u0102\u0103\u00071\u0002\u0002\u0103\u0104\u0007",
    "1\u0002\u0002\u0104\u0108\u0003\u0002\u0002\u0002\u0105\u0107\n\f\u0002",
    "\u0002\u0106\u0105\u0003\u0002\u0002\u0002\u0107\u010a\u0003\u0002\u0002",
    "\u0002\u0108\u0106\u0003\u0002\u0002\u0002\u0108\u0109\u0003\u0002\u0002",
    "\u0002\u0109\u010c\u0003\u0002\u0002\u0002\u010a\u0108\u0003\u0002\u0002",
    "\u0002\u010b\u010d\u0007\u000f\u0002\u0002\u010c\u010b\u0003\u0002\u0002",
    "\u0002\u010c\u010d\u0003\u0002\u0002\u0002\u010d\u010e\u0003\u0002\u0002",
    "\u0002\u010e\u010f\u0007\f\u0002\u0002\u010f\u0110\u0003\u0002\u0002",
    "\u0002\u0110\u0111\b$\u0002\u0002\u0111H\u0003\u0002\u0002\u0002\u0112",
    "\u0113\u00071\u0002\u0002\u0113\u0114\u00071\u0002\u0002\u0114\u0118",
    "\u0003\u0002\u0002\u0002\u0115\u0117\n\f\u0002\u0002\u0116\u0115\u0003",
    "\u0002\u0002\u0002\u0117\u011a\u0003\u0002\u0002\u0002\u0118\u0116\u0003",
    "\u0002\u0002\u0002\u0118\u0119\u0003\u0002\u0002\u0002\u0119\u011b\u0003",
    "\u0002\u0002\u0002\u011a\u0118\u0003\u0002\u0002\u0002\u011b\u011c\b",
    "%\u0002\u0002\u011cJ\u0003\u0002\u0002\u0002\u001c\u0002RVXx\u007f\u0084",
    "\u009d\u00a2\u00aa\u00b0\u00b4\u00bb\u00bd\u00cb\u00d2\u00d7\u00dd\u00e3",
    "\u00e5\u00e9\u00ee\u00fa\u0108\u010c\u0118\u0003\b\u0002\u0002"].join("");


var atn = new antlr4.atn.ATNDeserializer().deserialize(serializedATN);

var decisionsToDFA = atn.decisionToState.map( function(ds, index) { return new antlr4.dfa.DFA(ds, index); });

function TPGrammarLexer(input) {
	antlr4.Lexer.call(this, input);
    this._interp = new antlr4.atn.LexerATNSimulator(this, atn, decisionsToDFA, new antlr4.PredictionContextCache());
    return this;
}

TPGrammarLexer.prototype = Object.create(antlr4.Lexer.prototype);
TPGrammarLexer.prototype.constructor = TPGrammarLexer;

TPGrammarLexer.EOF = antlr4.Token.EOF;
TPGrammarLexer.T__0 = 1;
TPGrammarLexer.T__1 = 2;
TPGrammarLexer.INVALID_LITERAL = 3;
TPGrammarLexer.IF = 4;
TPGrammarLexer.ELSE = 5;
TPGrammarLexer.EQ_DEF = 6;
TPGrammarLexer.OR = 7;
TPGrammarLexer.AND = 8;
TPGrammarLexer.NOT = 9;
TPGrammarLexer.EQ = 10;
TPGrammarLexer.LT = 11;
TPGrammarLexer.LE = 12;
TPGrammarLexer.GT = 13;
TPGrammarLexer.GE = 14;
TPGrammarLexer.NEQ = 15;
TPGrammarLexer.CONCAT = 16;
TPGrammarLexer.MINUSMINUS = 17;
TPGrammarLexer.MUL = 18;
TPGrammarLexer.DIV = 19;
TPGrammarLexer.PLUS = 20;
TPGrammarLexer.MINUS = 21;
TPGrammarLexer.BOOLEAN = 22;
TPGrammarLexer.INTEGER = 23;
TPGrammarLexer.FLOAT = 24;
TPGrammarLexer.ID = 25;
TPGrammarLexer.WS = 26;
TPGrammarLexer.COMMENT = 27;
TPGrammarLexer.LINE_COMMENT = 28;
TPGrammarLexer.LINE_COMMENT_EOF = 29;

TPGrammarLexer.prototype.channelNames = [ "DEFAULT_TOKEN_CHANNEL", "HIDDEN" ];

TPGrammarLexer.prototype.modeNames = [ "DEFAULT_MODE" ];

TPGrammarLexer.prototype.literalNames = [ null, "'('", "')'", null, "'if'", 
                                          "'else'", "'='", "'or'", "'and'", 
                                          "'not'", "'=='", "'<'", null, 
                                          "'>'", null, null, "'++'", "'--'", 
                                          null, null, "'+'" ];

TPGrammarLexer.prototype.symbolicNames = [ null, null, null, "INVALID_LITERAL", 
                                           "IF", "ELSE", "EQ_DEF", "OR", 
                                           "AND", "NOT", "EQ", "LT", "LE", 
                                           "GT", "GE", "NEQ", "CONCAT", 
                                           "MINUSMINUS", "MUL", "DIV", "PLUS", 
                                           "MINUS", "BOOLEAN", "INTEGER", 
                                           "FLOAT", "ID", "WS", "COMMENT", 
                                           "LINE_COMMENT", "LINE_COMMENT_EOF" ];

TPGrammarLexer.prototype.ruleNames = [ "T__0", "T__1", "INVALID_LITERAL", 
                                       "IF", "ELSE", "EQ_DEF", "OR", "AND", 
                                       "NOT", "EQ", "LT", "LE", "GT", "GE", 
                                       "NEQ", "CONCAT", "MINUSMINUS", "MUL", 
                                       "DIV", "PLUS", "MINUS", "BOOLEAN", 
                                       "INTEGER", "FLOAT", "ID", "IDENTIFIER", 
                                       "J_LETTER", "DIGIT", "HEX_DIGIT", 
                                       "BIN_DIGIT", "DEC_SIGNIFICANT", "DEC_EXPONENT", 
                                       "WS", "COMMENT", "LINE_COMMENT", 
                                       "LINE_COMMENT_EOF" ];

TPGrammarLexer.prototype.grammarFileName = "TPGrammar.g4";



exports.TPGrammarLexer = TPGrammarLexer;
